### PR TITLE
feat: Improved error handling when running tests

### PR DESCRIPTION
**Pull request by mull**
Since knowing where errors are coming from is quite hard this commit makes sure to try and report whether it's from Kubernetes or an error in the test itself.

A few improvements to the K8s and testing code in general to add more tests for errors and more granular error reporting.

### CODE REVIEW

File: packages/tests/testWorker.js [hunk 1]
```diff
@@ -1,30 +1,64 @@
...
+      await job.log(`TestError: ${err}`)
+    } else {
+      await job.log(`WorkerError: ${err}`)
+    }
+
+    console.log(`Error: ${err}`)
     throw err;
   } finally {
-    deleteTest(podName)
+    await deleteTest(podName)
```
Comment (mull) [inline packages/tests/testWorker.js:62]: This seems to fix an edge case/race condition with lingering pods in the test namespace

**Review by izabellalarsson — APPROVED** [review]
lgtm

File: packages/tests/testWorker.js (hunk 1 above)
Comment (irony) [reply packages/tests/testWorker.js:62]: Makes sense. The job is a promise and when the promise is done it's not going to wait until the deletion is completed and an error in the promise wouldn't be catched.

File: packages/tests/testWorker.js [hunk 2]
```diff
@@ -1,30 +1,64 @@
...
+  const {passed, details} = response
+  return {passed, details}
+}
+
 module.exports = (test_name, image, connection, resultQueue) => new Worker(test_name, async (job) => {
   console.log(`Starting ${test_name}`)
   await job.log(`Starting ${job.name}`)
   const {id, data: { arguments, test_run_id }} = job
   const podName = `${test_name}-${test_run_id}`
+  const containerArguments = [
```
[the remark this answers is not in the document]
Comment (irony) [reply packages/tests/testWorker.js:33]: Objects keep their insertion orders for string keys. Yes.

**Review by irony — COMMENTED** [review]
Looks good.
